### PR TITLE
Move project to Typescript

### LINKED ISSUE #6: Move project to Typescript

### PR DESCRIPTION
**Pull request by sorokinvj**
This PR introduces breaking changes:

- all code except configs are now in .ts, .tsx files
- changed tailwind CSS config
- all static props are now revalidated, which means we could add new products, categories, and collections after build and have them rendered despite the static nature of these pages
- success page with the order are now clearing the cart
- hygraph config is deleted. It was used only in Footer to render language dropdown, its refactored using next.config i18n property
- new SEO description and title, no SEO for cart page
- new hook - `useHasMounted` to deal with a re-hydration error that appears because of `react-use-cart` implementation - it uses localStorage to keep cart data and update the DOM. Unfortunately, server and client DOM must be identical on the first render, so we use `useHasMounted` to define if the component mounted or not and if not - return null. This is a pattern described in an excellent article [The Perils of Rehydration](https://www.joshwcomeau.com/react/the-perils-of-rehydration/#introduction)
- quantity cannot be changed now on stripe checkout (this option was turned on before)

closes #6

### CODE REVIEW

File: pages/products/[slug].tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      ...products.map((product: any) => ({
+        params: { slug: product.slug },
+        locale
+      }))
+    ]
+  }
+
+  return {
+    paths,
+    fallback: 'blocking'
```
Comment (sorokinvj) [inline pages/products/[slug].tsx:44]: @valieff this is how we re-build the product page when the `slug` (product id) is not the one that was known during the build

File: pages/products/[slug].tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  })
+
+  console.log(product)
+
+  return {
+    props: {
+      product,
+      ...pageData
+    },
+    revalidate: 1
```
Comment (sorokinvj) [inline pages/products/[slug].tsx:60]: this means re-validating the page data in 1 sec - https://vercel.com/docs/concepts/next.js/incremental-static-regeneration

File: pages/products/[slug].tsx (hunk 1 above)
Comment (sorokinvj) [reply pages/products/[slug].tsx:44]: https://vercel.com/docs/concepts/next.js/incremental-static-regeneration

**Comment by valieff** [comment]
pr looks good to me!

**Review by valieff — APPROVED** [review]
looks great!

**Pull request merged**